kt/godot-runtime/src/main/kotlin/godot/core/pool/PoolRealArray.kt

```diff
@@ -0,0 +1,172 @@
+@file:Suppress("PackageDirectoryMismatch")
+
+package godot.core
+
+import godot.util.IndexedIterator
+import godot.util.RealT
+import godot.util.VoidPtr
+import godot.util.realTVariantType
+
+@Suppress("MemberVisibilityCanBePrivate", "unused")
+class PoolRealArray : NativeCoreType, Iterable<RealT> {

```

Comment:
Why are the pool arrays not inheriting mutable collection? just because of the function naming for append and so on? or does it have another reason as well?

Reply:
No particular reason, I just kinda implemented Itererable and didn't think about using a more specialized interface.

Reply:
shouldn't we then not use mutable collection?
Then one can use all of the kotlin collection function

Reply:
Maybe we can do it after merging this as an enhancement ? This would allow us to split last tasks for alpha.

Reply:
Ok for me